src/client/serialize/daos_serialize.c

```diff
@@ -5,13 +5,77 @@
 #include <daos_cont.h>
 
 #define SERIALIZE_ATTR_DSET "User Attributes"
+#define OID_ARR_SIZE 8
+
+/* TODO: mostly needs to be removed with dynamic key buffers */
+#define ENUM_KEY_BUF		128 /* size of each dkey/akey */
+#define ENUM_LARGE_KEY_BUF	(512 * 1024) /* 512k large key */
+#define ENUM_DESC_NR		5 /* number of keys/records returned by enum */
+#define ENUM_DESC_BUF		512 /* all keys/records returned by enum */
+#define ATTR_NAME_LEN		128
+#define SERIALIZE_VERSION	0.0
+
+#if defined(DAOS_API_VERSION_MAJOR) && defined(DAOS_API_VERSION_MINOR)
+#define CHECK_DAOS_API_VERSION(major, minor)                            \
+	((DAOS_API_VERSION_MAJOR > (major))                             \
+	|| (DAOS_API_VERSION_MAJOR == (major) && DAOS_API_VERSION_MINOR >= (minor)))
+#else
+#define CHECK_DAOS_API_VERSION(major, minor) 0
+#endif
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
 
 /* for user attr dataset */
 struct usr_attr {
 	char	*attr_name;
 	hvl_t	attr_val;
 };
 
+/* for oid dataset */
+struct oid {
+	uint64_t oid_hi;
+	uint64_t oid_low;
+	uint64_t dkey_offset;
+};
+
+/* for dkey dataset */
+struct dkey {
+	/* array of vlen structure */
+	hvl_t dkey_val;
+	uint64_t akey_offset;
+	/* for for kv values that can be
+	 * written with daos_kv.h
+	 */
+	hvl_t rec_kv_val;
+};
+
+/* for akey dataset */
+struct akey {
+	/* array of vlen structure */
+	hvl_t akey_val;
+	uint64_t rec_dset_id;
+	hvl_t rec_single_val;
+};
+
+struct hdf5_args {
```

Comment:
also rename this to something like
dsr_h5_args (daos_serialize)

Reply:
Done